Test/Unit/Helper/Webhook/AuthorisationWebhookHandlerTest.php

```diff
@@ -1,130 +1,363 @@
+        $paymentMock = $this->createMock(Order::class);
+        $storeId = 1;
+        $this->orderMock->method('getStoreId')->willReturn($storeId);
+        $this->orderMock->method('getPayment')->willReturn($paymentMock);
+
+        $this->orderHelperMock->expects($this->once())
+            ->method('setPrePaymentAuthorized')->willReturn($this->orderMock);
+        $this->orderHelperMock->expects($this->once())
+            ->method('updatePaymentDetails');
+        $this->orderHelperMock->expects($this->once())
+            ->method('sendOrderMail');
+        $this->orderHelperMock->expects($this->once())
+            ->method('finalizeOrder')->willReturn($this->orderMock);
+
+        $paymentMethodsMock = $this->createConfiguredMock(
+            PaymentMethods::class,
+            [
+                'isAutoCapture' => true
+            ]
+        );
+
+        $authorisationWebhookHandler = $this->createAuthorisationWebhookHandler(
+            $this->adyenOrderPaymentMock,
+            $this->orderHelperMock,
             null,
             null,
             null,
             $mockChargedCurrency,
             null,
             null,
-            $mockPaymentMethodsHelper
+            $paymentMethodsMock
+        );
+
+        // Invoke the private method
+        $handleSuccessfulAuthorisationMethod = $this->getPrivateMethod(
+            AuthorisationWebhookHandler::class,
+            'handleSuccessfulAuthorisation'
         );
 
-        $transitionState = PaymentStates::STATE_FAILED;
-        $result = $handler->handleWebhook($order, $notification, $transitionState);
+        $result = $handleSuccessfulAuthorisationMethod->invokeArgs($authorisationWebhookHandler, [$this->orderMock, $this->notificationMock]);
+
+        // Assert the result
+        $this->assertInstanceOf(Order::class, $result);
+    }
 
-        // Assert that handleFailedAuthorisation was called and handleSuccessfulAuthorisation wasn't called
-        $this->assertNotNull($result);
-        $this->assertEquals($order->toArray(), $result->toArray());
+    public function testHandleFailedAuthorisation()
+    {
+        $this->orderMock->expects($this->atLeastOnce())
+            ->method('getData')
+            ->withConsecutive(
+                ['adyen_notification_event_code'],
+                ['adyen_notification_payment_captured']
+            )
+            ->willReturnCallback(function ($arg) {
+                if ($arg === 'adyen_notification_event_code') {
+                    return 'AUTHORISATION : TRUE';
+                } elseif ($arg === 'adyen_notification_payment_captured') {
+                    return false;
+                }
+            });
 
+        // Create an instance of AuthorisationWebhookHandler
+        $webhookHandler = $this->createAuthorisationWebhookHandler(
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null
```

Comment:
You can skip sending values for optional arguments. They are already null by default.